src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -463,7 +463,9 @@ public PossiblyConditionalState Clone()
                                                 // The expression in the tuple could not be assigned a slot (for example, `a?.b`),
                                                 // so we had to create a slot for the tuple element instead.
                                                 // We'll remember that so that we can apply any learnings to the expression.
+#pragma warning disable CA1854 //Prefer a 'TryGetValue' call over a Dictionary indexer access guarded by a 'ContainsKey' check to avoid double lookup
```

Comment:
The indexAccess is in the else path `Debug.Assert(originalInputMap[outputSlot] == ...`, 
here i could replace it with a `TryAdd` since the overhead of cast and array access is probably small enough.


If there is access and add like
```c#
if(!x.Contains(y))
    x.Add(y,z);
else
    Use(x[y]);
```
or
```c#
if(!x.Contains(y))
    x.Add(y, new());

x[y].Add(z);
```
than i think `TryGet` is better than `TryAdd`, since in most cases the add is only being used as an init.